src/common/utils.cc

```diff
@@ -73,32 +125,48 @@ int PinToCore(int core_id) {
 
 void PinToCoreWithOffset(ThreadType thread_type, int core_offset, int thread_id,
                          bool verbose) {
+  std::scoped_lock lock(pin_core_mutex);
+
   if (kEnableThreadPinning == true) {
-    int actual_core_id = core_offset + thread_id;
-    int num_cores = sysconf(_SC_NPROCESSORS_ONLN);
+    size_t requested_core = core_offset + thread_id;
+    const int num_cores = sysconf(_SC_NPROCESSORS_ONLN);
 
-    /* Reserve core 0 for kernel threads */
-    if (actual_core_id >= num_cores) {
-      actual_core_id = (actual_core_id % num_cores) + 1;
+    if (kEnableThreadWrapping) {
+      requested_core = requested_core % num_cores;
```

Comment:
I think there is no need to have kEnableThreadWrapping here. The original Line 80-82 is used for the case where NIC is installed on the second socket so core offset starts from the second socket (e.g., the offset is 18 for a 36 core 2-socket server), and we want to use the cores on the first socket when thread number is large. 

We want to let this function handle this case by default (kEnableThreadWrapping should always be true). 

Since we also want to skip core 0 for the core assignment, we should increment requested_core by 1 after getting the remainder.